### ISSUE TITLE
result is not array on getAccountDetails using eparcel account number

### ISSUE THREAD

**Issue by tashantukin** [joshbmarshall/auspost#4]
Hi, i tried to use an eparcel account number and it seems to add an 220F <result> on the response. am I missing something?

Thanks,

**Comment by joshbmarshall** [comment]
are you using the api from the Auspost Business account?

**Comment by tashantukin** [comment]
Hi @joshbmarshall thanks for replying.  I am not really sure. how can I check that? ive tried to use the endpoint from postman and it works fine.

**Comment by joshbmarshall** [comment]
This library uses the endpoints at digitalapi.auspost.com.au - does that match what you're using?

**Comment by tashantukin** [comment]
this is the response from Postman, I think you are right about the austpost
```
{
    "account_number": "2001191933",
    "name": "DEMO - AusPost",
    "valid_from": "2017-11-01",
    "valid_to": "2999-01-01",
    "expired": false,
    "addresses": [
        {
            "type": "MERCHANT_LOCATION",
            "lines": [
                "1 Main Street"
            ],
            "suburb": "MELBOURNE",
            "postcode": "3000",
            "state": "VIC",
            "country": "AU"
        }
    ],
```

**Comment by tashantukin** [comment]
![image](https://github.com/joshbmarshall/auspost/assets/32629251/c0c15f48-9b81-4079-a940-4a6b268435ef)

**Comment by joshbmarshall** [comment]
Have you set the test mode correctly?

**Comment by tashantukin** [comment]
yes, I have set it to` true`, this is where is it giving a non array response, with 220F,
![image](https://github.com/joshbmarshall/auspost/assets/32629251/65b141c9-9872-4fc2-9987-660ed061b321)
![image](https://github.com/joshbmarshall/auspost/assets/32629251/d154ca0e-f1ce-4c19-b3ae-06fd13eccb9a)

**Comment by joshbmarshall** [comment]
can you debug up the chain, ie what is $this->getResponse() , what is coming back from Auspost? see if you can find where the 220f comes in

**Comment by tashantukin** [comment]
hi @joshbmarshall  i think it is coming from here, sorry I am not familiar with sockets
![image](https://github.com/joshbmarshall/auspost/assets/32629251/3aedd6c8-20bf-4ea0-b471-483cf66c7f13)

**Comment by tashantukin** [comment]
hi @joshbmarshall in any case, do you have any idea why that 220f is showing up on the response? thanks.

**Comment by joshbmarshall** [comment]
The only way I can see it in the response is if the api was sending it to you. Have you tried a different php version or platform?

**Comment by jp135i** [comment]
Hi Josh,

Thanks for all your hard work on this library.

I am seeing the same sort of behaviour but with different prefix characters - I have seen both "2001" and "399d" and then what looks like a correct response, however not the full response as it seems to be cut off.

Here is my response in php (some specific details edited) =>
"399d
{"account_number":"2010000000","name":"DEMO - AusPost","valid_from":"2017-11-01","valid_to":"2999-12-31","expired":false,"addresses":[{"type":"MERCHANT_LOCATION","lines":["1 Main Street"],"suburb":"MELBOURNE","postcode":"3000","state":"VIC","country":"AU"}],"details":{"is_sscc_non_standard_barcode_enabled_international":"false","email_address":"emailaddress@company","is_location":"false","is_sscc_non_standard_barcode_enabled":"false","is_on_demand":"false","is_sender_distribution_state_override":"false","contact_number":"","account_source":"AUSPOST"},"postage_products":[{"type":"EXPRESS EPARCEL ID&V 2","group":"Express Post","product_id":"XID2","contract":{"valid_from":"2017-11-01","valid_to":"2999-12-31","expired":false,"volumetric_pricing":false,"max_item_count":0,"cubing_factor":250},"authority_to_leave_threshold":500.0,"features":{"TRANSIT_COVER":{"type":"TRANSIT_COVER","attributes":{"rate":1,"included_cover":0,"maximum_cover":5000},"price":{"calculated_gst":0.00,"calculated_price":0.00,"calcula"

If I use ThunderClient in vscode if don't receive the prefix characters and get a full correct response.

I am using PHP Version 7.4.3-4ubuntu2.19

Can I provide you any further information to help resolve this?

Cheers,

**Comment by joshbmarshall** [comment]
Hi folks, the code was working for me on all the versions of PHP I tested on 7.2 through to 8.2

The original code using sockets is what AusPost recommended when I wrote the library. But it's not very nice.

So I just rewrote these parts to use curl and added a unit test to help with testing.

Can you confirm whether this gets the library working for you? If so I'll create a new release.

**Comment by joshbmarshall** [comment]
Actually have just release 1.2.0 - the code is much nicer. I'd like your feedback if it gets your issue resolved
